libgalois/include/galois/analytics/sssp/sssp.h

```diff
@@ -134,412 +134,32 @@ using SsspEdgeWeight = galois::PODProperty<Weight>;
-            }
-          },
-          galois::steal(), galois::loopname("Update"));
-
-    } while (changed.reduce());
-
-    galois::ReportStatSingle("SSSP-Topo", "rounds", rounds);
-  }
-
-public:
-  galois::Result<void> SSSP(Graph& graph, size_t start_node, SsspPlan plan) {
-    if (start_node >= graph.size()) {
-      return galois::ErrorCode::InvalidArgument;
-    }
-
-    auto it = graph.begin();
-    std::advance(it, start_node);
-    typename Graph::Node source = *it;
-
-    size_t approxNodeData = graph.size() * 64;
-    galois::Prealloc(1, approxNodeData);
-
-    galois::do_all(
-        galois::iterate(graph), [&graph](const typename Graph::Node& n) {
-          graph.template GetData<NodeDistance>(n) = kDistanceInfinity;
-        });
-
-    graph.template GetData<NodeDistance>(source) = 0;
-
-    galois::StatTimer execTime("SSSP");
-    execTime.start();
+GALOIS_EXPORT Result<bool> SsspValidate(
+    graphs::PropertyFileGraph* pfg, size_t start_node,
+    const std::string& edge_weight_property_name,
+    const std::string& output_property_name);
 
-    if (plan.algorithm() == SsspPlan::kAutomatic) {
-      plan = SsspPlan::Automatic(&graph.GetPropertyFileGraph());
-    }
+struct GALOIS_EXPORT SsspStatistics {
+  /// The maximum distance across all nodes.
+  double max_distance;
+  /// The sum of all node distances.
+  double total_distance;
+  /// The number of nodes reachable from the source node.
+  uint32_t n_reached_nodes;
 
-    switch (plan.algorithm()) {
-    case SsspPlan::kDeltaTile:
-      DeltaStepAlgo<SrcEdgeTile>(
-          &graph, source, SrcEdgeTilePushWrap{&graph, *this}, TileRangeFn(),
-          plan.delta());
-      break;
-    case SsspPlan::kDeltaStep:
-      DeltaStepAlgo<UpdateRequest>(
-          &graph, source, ReqPushWrap(), OutEdgeRangeFn{&graph}, plan.delta());
-      break;
-    case SsspPlan::kSerialDeltaTile:
-      SerDeltaAlgo<SrcEdgeTile>(
-          &graph, source, SrcEdgeTilePushWrap{&graph, *this}, TileRangeFn(),
-          plan.delta());
-      break;
-    case SsspPlan::kSerialDelta:
-      SerDeltaAlgo<UpdateRequest>(
-          &graph, source, ReqPushWrap(), OutEdgeRangeFn{&graph}, plan.delta());
-      break;
-    case SsspPlan::kDijkstraTile:
-      DijkstraAlgo<SrcEdgeTile>(
-          &graph, source, SrcEdgeTilePushWrap{&graph, *this}, TileRangeFn());
-      break;
-    case SsspPlan::kDijkstra:
-      DijkstraAlgo<UpdateRequest>(
-          &graph, source, ReqPushWrap(), OutEdgeRangeFn{&graph});
-      break;
-    case SsspPlan::kTopo:
-      TopoAlgo(&graph, source);
-      break;
-    case SsspPlan::kTopoTile:
-      TopoTileAlgo(&graph, source);
-      break;
-    case SsspPlan::kDeltaStepBarrier:
-      DeltaStepAlgo<UpdateRequest, OBIMBarrier>(
-          &graph, source, ReqPushWrap(), OutEdgeRangeFn{&graph}, plan.delta());
-      break;
-    default:
-      return galois::ErrorCode::InvalidArgument;
-    }
+  float average_distance() { return total_distance / n_reached_nodes; }
```

Comment:
float(total_distance)?

Reply:
Actually I think it should return double.